File: src/main/java/guru/xgm/language/angelscript/formatting/block/utils/AngelScriptBlockSettings.java
package guru.xgm.language.angelscript.formatting.block.utils;

import com.intellij.formatting.FormattingContext;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.CommonCodeStyleSettings;
import guru.xgm.language.angelscript.formatting.AngelScriptCodeStyleSettings;
import guru.xgm.language.angelscript.lang.AngelScriptLanguage;

public class AngelScriptBlockSettings {
    public AngelScriptBlockSettings(FormattingContext formattingContext) {
        code = formattingContext.getCodeStyleSettings();
        common = code.getCommonSettings(AngelScriptLanguage.INSTANCE.getID());
        custom = code.getCustomSettings(AngelScriptCodeStyleSettings.class);
    }

    public final CodeStyleSettings code;
    public final CommonCodeStyleSettings common;
    public final AngelScriptCodeStyleSettings custom;
}

File: src/main/kotlin/guru/xgm/language/angelscript/formatting/block/utils/AngelScriptBlockSettings.kt
package guru.xgm.language.angelscript.formatting.block.utils

import com.intellij.formatting.FormattingContext
import com.intellij.psi.codeStyle.CodeStyleSettings
import com.intellij.psi.codeStyle.CommonCodeStyleSettings
import guru.xgm.language.angelscript.formatting.AngelScriptCodeStyleSettings
import guru.xgm.language.angelscript.lang.AngelScriptLanguage

class AngelScriptBlockSettings(formattingContext: FormattingContext) {
    val code: CodeStyleSettings = formattingContext.codeStyleSettings
    val common: CommonCodeStyleSettings = code.getCommonSettings(AngelScriptLanguage.instance.id)
    val custom: AngelScriptCodeStyleSettings = code.getCustomSettings(AngelScriptCodeStyleSettings::class.java)
}
